### ISSUE TITLE
Why is Slider an "output" that produces "aw" messages?

### ISSUE THREAD

**Issue by rwaldron** [blynkkk/blynk-server#130]
If a slider is a sensor, and a sensor should be read as input...

- why does the Slider widget produce "aw" (presumable short for "analog write") messages?
- why is the Slider widget limited to _digital pwm_ pins? It's an analog input (ie. variable resistance)

**Comment by rwaldron** [comment]
Same question for Joystick. 


I think I understand how/why this approach might seem interesting, but it's unintuitive.

**Comment by doom369** [comment]
> If a slider is a sensor

Slider is not a sensor =). It is controller. But Blynk allows to use it in both ways.

> It's an analog input 

@blynkkk maybe can ansewer.

**Comment by doom369** [comment]
I'll try to answer again as I thought this question is from another repository.

So initially slider was designed as control for analog input (as you mentioned, for example change LED fading). After Blynk was launched we got a lot requests like "can I set slider to some value from hardware?". For example when I turn potentiometer I want to move slider to appropriate position. So we thought - "why not?" and did it. So now slider work in both ways and we did the same for all widgets Blynk has.

Regrding aw, pwm - I'm not ready to answer on this questions. I think https://github.com/blynkkk/blynk-library is correct place to ask it. @vshymanskyy could explain.
